### ISSUE TITLE
i18n_select: option "Select taxonomy terms by language" is missing

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/i18n#105]
While I was trying ([see related Zulip thread](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Show.20only.20translated.20terms.20in.20Views.20exposed.20filter/near/377014911)) to configure an exposed Views filter with the goal to show only taxonomy terms of the current user's language, I enabled the `i18n_select` submodule. That didn't help me with the exposed filter, probably because the appropriate option "Select taxonomy terms by language" on the settings page of the submodule version is missing in Backdrop.

The Drupal module has _two_ options under "Content to filter by language", if "Taxonomy translation" is enabled:

- Select nodes by language
- Select taxonomy terms by language

The Backdrop version shows only the first option, even if "Taxonomy translation" is enabled.

**Screenshot from the [Drupal docs](https://www.drupal.org/node/1279512):**

![image](https://github.com/backdrop-contrib/i18n/assets/14188307/abf1c7e3-21c7-4c17-b9b4-c458bf8b7813)

**Screenshot from Backdrop** (admin/config/regional/i18n/select):

![image](https://github.com/backdrop-contrib/i18n/assets/14188307/05deb90b-f1f1-4dde-8bc4-fd4a9193fbe9)

**Comment by indigoxela** [comment]
@olafgrabienski many thanks for taking the time to report.

As a first step I'm trying to recreate the setup in D7. It seems to be related to the Variables module, which doesn't exist in B (as the whole concept doesn't exist).

**Comment by indigoxela** [comment]
A-ha, definitely a bug. :wink:

Getting the form item to show up is the easy part. Not sure, if that already has influence on results. (The update hook is still missing). @olafgrabienski do you have a test site, where you can safely apply patches?

**Comment by olafgrabienski** [comment]
@indigoxela Thank you very much for looking into the issue! I'll configure a site to test the PR later today.

**Comment by indigoxela** [comment]
How to test:

- Apply the patch
- Run the update
- Go to /admin/config/regional/i18n/select and turn "Select taxonomy terms by language" on
- Make sure, you're using a select list for your exposed filter and *not* autocomplete

Some caveats:

- This setting is a global one, which applies to ALL taxonomy related db queries. Make sure, that other views or pages don't behave weird
- This setting isn't applied on the admin/* path by default, so you don't see the effect in the views preview (and other admin paths)

**Comment by olafgrabienski** [comment]
Tested the PR roughly, works for me!

On my test site, a select list displayed terms in both languages, as described in the issue report. After applying the PR and checking the settings at `admin/config/regional/i18n/select`, the list displayed only terms in the 'current user's language', as expected.

Regarding the new file [i18n_select.install](https://github.com/backdrop-contrib/i18n/pull/106/files#diff-1228edf7f7947c92018e0ba7b727410f3c66d01e2d2922457888920bed699b93), I did not test to install the patched module (and guess I won't have time the next two weeks).
